ecc/bls12-377/fp/element_mul_amd64.s

```diff
@@ -1,4 +1,4 @@
-// +build !amd64_adx
+// +build !purego
```

Comment:
I'm confused by `amd64_adx` being replaced with `purego`. I thought that when ADX was available we definitely used assembly and hence it wouldn't be pure Go.

Reply:
Is what's happening the following:

Previously we had two sets of x64 assembly, one for when ADX is available and one when not. Turns out assembly is not quite worth it unless ADX is available hence the `purego` flag is set unless ADX is available.

Reply:
`purego` is a wider Golang community conventioon, to avoid the use of assembly at all. 

The previous `amd64_adx` was removing the instructions to actually check the presence of the ADX instructions at run time, which can result in a minor 2-4% speed up.